source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -140,9 +141,27 @@ void AuthenticatorImpl::startVerify() {
   curr_token_ = std::move(tokens_.back());
   tokens_.pop_back();
 
-  jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
-  ENVOY_LOG(debug, "{}: Parse Jwt {}", name(), curr_token_->token());
-  Status status = jwt_->parseFromString(curr_token_->token());
+  if (provider_) {
+    jwks_data_ = jwks_cache_.findByProvider(provider_.value());
+    if (jwks_data_->getJwtProvider().enable_token_cache()) {
+      jwt_ = jwks_data_->getTokenCache().find(curr_token_->token());
+      if (jwt_) {
+        use_cache_jwt_ = true;
+        jwks_data_->getTokenCache().release(curr_token_->token(), jwt_);
+      }
+    }
+  }
+  Status status;
+  if (use_cache_jwt_ && provider_) {
```

Comment:
not need to check provider_,  use_cache_jwt_ is true inplies it

Reply:
done.